compiler/rustc_typeck/src/structured_errors/wrong_number_of_generic_args.rs

```diff
@@ -626,32 +626,42 @@ impl<'a, 'tcx> WrongNumberOfGenericArgs<'a, 'tcx> {
         let remove_entire_generics = num_redundant_args >= self.gen_args.args.len();
 
         let remove_lifetime_args = |err: &mut DiagnosticBuilder<'_>| {
-            let idx_first_redundant_lt_args = self.num_expected_lifetime_args();
+            let mut lt_arg_indices = Vec::new();
+            let mut found_redundant = false;
+            for (i, arg) in self.gen_args.args.iter().enumerate() {
+                if let hir::GenericArg::Lifetime(_) = arg {
+                    if i > self.num_expected_lifetime_args() - 1 {
+                        found_redundant = true;
+                    }
+                    lt_arg_indices.push(i);
+                } else if found_redundant == true {
+                    // Argument which is redundant and separated like this `'c`
+                    // is not included to avoid including `Bar` in span.
+                    // ```
+                    // type Foo<'a, T> = &'a T;
+                    // let _: Foo<'a, 'b, Bar, 'c>;
+                    // ```
+                    break;
+                }
+            }
+
+            let idx_first_redundant_lt_args = lt_arg_indices[self.num_expected_lifetime_args()];
             let span_lo_redundant_lt_args =
                 self.gen_args.args[idx_first_redundant_lt_args].span().shrink_to_lo();
-            let span_hi_redundant_lt_args = self.gen_args.args
-                [idx_first_redundant_lt_args + num_redundant_lt_args - 1]
-                .span()
-                .shrink_to_hi();
-            let eat_comma =
-                idx_first_redundant_lt_args + num_redundant_lt_args - 1 != self.gen_args.args.len();
-
-            let span_redundant_lt_args = if eat_comma {
-                let span_hi = self.gen_args.args
-                    [idx_first_redundant_lt_args + num_redundant_lt_args - 1]
-                    .span()
-                    .shrink_to_hi();
-                span_lo_redundant_lt_args.to(span_hi)
-            } else {
-                span_lo_redundant_lt_args.to(span_hi_redundant_lt_args)
-            };
+            let idx_last_redundant_lt_args = lt_arg_indices[lt_arg_indices.len() - 1];
+            let span_hi_redundant_lt_args =
+                self.gen_args.args[idx_last_redundant_lt_args].span().shrink_to_hi();
+
+            let span_redundant_lt_args = span_lo_redundant_lt_args.to(span_hi_redundant_lt_args);
             debug!("span_redundant_lt_args: {:?}", span_redundant_lt_args);
 
+            let num_redundant_lifetime_args =
+                lt_arg_indices.len() - self.num_expected_lifetime_args();
             let msg_lifetimes = format!(
                 "remove {} {} argument{}",
-                if num_redundant_args == 1 { "this" } else { "these" },
+                if num_redundant_lifetime_args == 1 { "this" } else { "these" },
                 "lifetime",
-                pluralize!(num_redundant_lt_args),
+                pluralize!(num_redundant_lifetime_args),
```

Comment:
```suggestion
                pluralize!(num_redundant_lt_args),
```
Nit: other variables above use `lt`, so renaming this variable alone seems inconsistent.